rust/src/dataset/scanner.rs

```diff
@@ -216,24 +216,37 @@ impl Scanner {
     pub fn schema(&self) -> Result<SchemaRef> {
         if self.nearest.as_ref().is_some() {
             let q = self.nearest.as_ref().unwrap();
-            let column: ArrowField = self
-                .dataset
-                .schema()
-                .field(q.column.as_str())
-                .ok_or_else(|| {
-                    Error::Schema(format!("Vector column {} not found in schema", q.column))
-                })?
-                .into();
+            let vector_schema = self.dataset.schema().project(&[&q.column])?;
             let score = ArrowField::new("score", Float32, false);
-            let score_schema = ArrowSchema::new(vec![column, score]);
-            let vector_search_columns = &Schema::try_from(&score_schema)?;
-            let merged = self.projections.merge(vector_search_columns);
+            let score_schema = ArrowSchema::new(vec![score]);
+
+            let merged = self
+                .projections
+                .merge(&vector_schema)
+                .merge(&Schema::try_from(&score_schema)?);
             Ok(SchemaRef::new(ArrowSchema::from(&merged)))
         } else {
             Ok(Arc::new(ArrowSchema::from(&self.projections)))
         }
     }
 
+    fn scanner_output_schema(&self) -> Result<Arc<Schema>> {
```

Comment:
Should this just be `Scanner::schema`?

Reply:
From the API aspect, `Scanner::schema` should just return the schema of output. So the contract can be built with other system components.

Reply:
Scanner::schema is the ArrowSchema and should not be used in Lance reader internals because that's why the field id's are messed up.

Reply:
I'll just make Scanner::schema call the other